rune/libenclave/internal/runtime/pal/skeleton/liberpal-skeleton.c

```diff
@@ -151,7 +151,16 @@ static bool encl_create(int dev_fd, unsigned long bin_size,
 	if (!base)
 		return false;
 
+/* If macro ZEROPAGE_MAPPING is defined, enclave will be mapped to
+ * zero address in process. It need to set sysctl vm.mmap_min_addr
+ * to zero firstly.
```

Comment:
Really? What happens if we unconditionally set the base address of SECS to 0 and **DON'T** set vm.mmap_min_addr to zero?

Reply:
It will crash on mmap(0, enclave_fd), If not set vm.mmap_min_addr, 0 address is not permitted to access.

Reply:
I don't think you need to do mmap(0, enclave_fd). You just keep the current mmap call.

Reply:
When calling ioctl(fd, SGX_IOC_ENCLAVE_CREATE), fill secs.base_addr to 0, and keep everything else, see what happens.

Reply:
INFO[0000] Initializing enclave runtime
INFO[0000] There is no enclave in cache pool
fatal error: unexpected signal during runtime execution
[signal SIGBUS: bus error code=0x2 addr=0x7f47d5178fff pc=0x7f47d518e3fe]

Reply:
secs.baseaddr and secs.size should present address range in enclave. So the mapping enclave's address should be in [secs.baseaddr, secs.baseaddr + secs.size -1]